godot-core/src/engine/gfile.rs

```diff
@@ -662,33 +662,25 @@ impl Read for GFile {
             return Ok(0);
         }
 
-        let mut read_bytes = 0;
-        while read_bytes < bytes_to_read {
-            buf[read_bytes] = self.fa.get_8();
-            read_bytes += 1;
-            self.check_error()?;
-        }
+        let gd_buffer = self.fa.get_buffer(bytes_to_read as i64);
+        buf[0..gd_buffer.len()].copy_from_slice(gd_buffer.as_slice());
+
+        self.check_error()?;
 
-        Ok(read_bytes)
+        Ok(gd_buffer.len())
```

Comment:
Could be `Ok(bytes_to_read)`.

You could add an `assert!` before, to verify the two are the same.